### DISCUSSION TITLE
browser.clearLocalStorage is not a function

### DISCUSSION THREAD

**Discussion by wangli-wangli** [webdriverio/webdriverio#6711] [Q&A] [answered]
I  write :
`browser.setLocalStorage("aa","bb");`
but  there is a wrong when running:
![image](https://user-images.githubusercontent.com/33982322/114696345-21348780-9d4f-11eb-9b7b-1145c26cbfc4.png)

**Comment by christian-bromann** [comment, marked as answer]
`setLocalStorage` is an outdated API command, we recommend to use the `execute` to access Web APIs, e.g.:

```js
browser.execute(() => localStorage.setItem('myCat', 'Tom'))
```

**Reply by wangli-wangli** [reply]
![image](https://user-images.githubusercontent.com/33982322/114696990-e7b04c00-9d4f-11eb-9a39-f586bf4fba62.png)
Name 'localstorage' not found

What should I do?

**Reply by christian-bromann** [reply]
try 

```js
browser.execute(() => window.localStorage.setItem('myCat', 'Tom'))
```

**Reply by wangli-wangli** [reply]
![image](https://user-images.githubusercontent.com/33982322/114802559-81b7d900-9dd0-11eb-8216-bbe7c91670be.png)

The name 'window' was not found.

**Reply by wangli-wangli** [reply]
![image](https://user-images.githubusercontent.com/33982322/114803883-dc523480-9dd2-11eb-9771-1b19152f71a4.png)
Windows only reports errors in. TS file, but not in JS file. But at run time, the content in JS file will not be saved, but the content in. TS will be run

**Reply by christian-bromann** [reply]
You will have to define `window` as a global in TypeScript because you switch from Node.js to a web environment.

**Reply by wangli-wangli** [reply]
Can you tell me how to write it?
